src/server/req_manager.c

```diff
@@ -2876,6 +2878,41 @@ make_host_addresses_list(char *phost, u_long **pul)
 		return 0;
 }
 
+/**
+ * @brief
+ * 		remove the cached ip addresses of a mom from the host tree and the ipaddrs tree
+ *
+ * @param[in]	pmom - ptr to the mom info
+ *
+ * @return	error code
+ * @retval	0			- no error
+ * @retval	PBS error	- error
+ */
+int
+remove_mom_ipaddresses_list(mominfo_t *pmom)
+{
+	/* take ipaddrs from ipaddrs cache tree */
+	if (hostaddr_tree != NULL) {
+		struct pul_store *tpul;
+		u_long *pul;
+
+		if ((tpul = (struct pul_store *) find_tree(hostaddr_tree, pmom->mi_host)) != NULL) {
+			pul = tpul->pul;
+			while (*pul) {
+					tdelete2(*pul, pmom->mi_port, &ipaddrs);
```

Comment:
Need to check pmom before we dereference?

Reply:
I feel we don't need to, since this is way deep in a stack of calls which all expect pmom to be valid. I can add a note to the description saying pmom is expected to be a valid pointer

Reply:
I disagree and think a NULL check should be put. Because the function might be called in other places where the call stack is not deep and `pmom` was not check. Even if there aren't such cases now in our code base there might be in the future.

Reply:
Agreed. Just double checking if this is one of the "sane" places where we can assume the value will never be NULL. Our increased use of tools like valgrind will help to identify any memory references that are out of bounds or dereferencing NULL pointers.